### PR TITLE
Download config.json for conversion script

### LINKED ISSUE #919: Small change in the scripts/download.py
Currently besides the safetensors and bin files the script only downloads:

download_files = ["tokenizer*", "generation_config.json"]

i think it would be beneficial to download all .json files such as config.json, vocab.json as they can be often necessary for converting these models to GGUF formats, what I noticed was that convert.py script from the llama cpp guys requires the config.json of the original repo and upon its absence it guesses the params which somehow doesn't fail the conversion but it fails at the inference load time.

I made the change to the above line to this all of my auto train to auto conversion pipelines are running without any issues. 

### Change suggested
download_files = ["tokenizer*", "*.json"]

with limited understanding of the entire repo not sure if this poses any risks with some of the config files generated by the training code like lit_config.json etc hence not improving this myself and rather suggesting it here.

### PR DESCRIPTION
**Pull request by rasbt**
Additionally downloads the `config.json` file, which is required if we want to convert Lit models back to HF and use the checkpoint via `transformers`. If this is merged, we can then simplify the tutorial in #959 

(Note that it's also downloading the `tokenizer_config.json` if it's present. Imho it's a super tiny file and doesn't hurt to download it if it exists; we may use it somewhere later in the future, who knows.)

Closes #919

### CODE REVIEW

**Comment by rasbt** [comment]
Redundant if we implement #923

**Review by awaelchli — APPROVED** [review]

File: scripts/download.py [hunk 1]
```diff
@@ -42,7 +42,7 @@ def download_from_hub(
             " https://huggingface.co/settings/tokens"
         )
 
-    download_files = ["tokenizer*", "generation_config.json"]
+    download_files = ["tokenizer*", "*config.json"]
```
Comment (carmocca) [inline scripts/download.py:45]: Lets be explicit instead of using *. We support many weights and there's no structure about what they can include

**Comment by rasbt** [comment]
Addressed your suggestion and updated the docs.

File: tutorials/convert_lit_models.md [hunk 2]
```diff
@@ -6,21 +6,21 @@ We provide a helpful script to convert models Lit-GPT models back to their equiv
 
 ```sh
 python scripts/convert_lit_checkpoint.py \
-    --checkpoint_path path/to/litgpt/model.pth \
-    --output_path out/converted.ckpt \
-    --config_path path/to/litgpt/config.json
+    --checkpoint_path checkpoints/repo_id/lit_model.pth \
+    --output_path output_path/converted.pth \
+    --config_path checkpoints/repo_id/config.json
```
Comment (carmocca) [inline tutorials/convert_lit_models.md:11]: I guess we could get rid of the `config_path` argument now

Comment (rasbt) [reply tutorials/convert_lit_models.md:11]: Let's do that in the PR where we copy the relevant files when creating checkpoint files.

**Pull request merged**